pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -120,96 +119,157 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(builder *ListenerBui
+					// This process typically yields multiple filter chain matches (with SNI) [if TLS is used]
+					tcpFilterChainOpts := make([]*filterChainOpts, 0)
+					for _, server := range serversForPort.Servers {
+						if gateway.IsTLSServer(server) && gateway.IsHTTPServer(server) {
+							routeName := mergedGateway.TLSServerInfo[server].RouteName
+							// This is a HTTPS server, where we are doing TLS termination. Build a http connection manager with TLS context
+							tcpFilterChainOpts = append(tcpFilterChainOpts, configgen.createGatewayHTTPFilterChainOpts(builder.node, server.Port, server,
+								routeName, proxyConfig, istionetworking.TransportProtocolTCP))
+							newFilterChains = append(newFilterChains, istionetworking.FilterChain{
+								ListenerProtocol:   istionetworking.ListenerProtocolHTTP,
+								IstioMutualGateway: server.Tls.Mode == networking.ServerTLSSettings_ISTIO_MUTUAL,
+							})
+						} else {
+							// This is the case of TCP or PASSTHROUGH.
+							tcpChainOpts := configgen.createGatewayTCPFilterChainOpts(builder.node, builder.push,
+								server, mergedGateway.GatewayNameForServer[server])
+							tcpFilterChainOpts = append(tcpFilterChainOpts, tcpChainOpts...)
+							for i := 0; i < len(tcpChainOpts); i++ {
+								newFilterChains = append(newFilterChains, istionetworking.FilterChain{
+									ListenerProtocol: istionetworking.ListenerProtocolTCP,
+								})
+							}
+						}
 					}
+
+					opts.filterChainOpts = tcpFilterChainOpts
+				}
+			case istionetworking.TransportProtocolQUIC:
+				// Currently, we just assume that QUIC is HTTP/3 although that does not
+				// have to be the case (it is just the most common case now, in the future
+				// we will support more cases)
+				quicFilterChainOpts := make([]*filterChainOpts, 0)
+				for _, server := range serversForPort.Servers {
+					log.Debugf("buildGatewayListeners: creating QUIC filter chain for port %d(%s:%s)",
+						server.GetPort().GetNumber(), server.GetPort().GetName(), server.GetPort().GetProtocol())
+					routeName := mergedGateway.TLSServerInfo[server].RouteName
+					if mergedGateway.TLSToQUICServerRouteMap[routeName] != "" {
+						routeName = mergedGateway.TLSToQUICServerRouteMap[routeName]
+					}
+					quicFilterChainOpts = append(quicFilterChainOpts, configgen.createGatewayHTTPFilterChainOpts(builder.node, server.Port, server,
+						routeName, proxyConfig, istionetworking.TransportProtocolQUIC))
+					newFilterChains = append(newFilterChains, istionetworking.FilterChain{
+						// Make sure that this is set to HTTP so that JWT and Authorization
+						// filters that are applied to HTTPS are also applied to this chain.
+						// Not doing so is a security hole as would allow bypassing auth.
+						ListenerProtocol:   istionetworking.ListenerProtocolHTTP,
+						TransportProtocol:  istionetworking.TransportProtocolQUIC,
+						IstioMutualGateway: false, // Currently, we don't support this. Revisit later
```

Comment:
just curious, why not?

Reply:
Simple answer - I didn't know much about that. I vaguely knew it is about mTLS. TBH searching for its usages, I still don't understand why it is there. So I set it to `false` and moved on so that I can focus on getting simple TLS to work.